### DISCUSSION TITLE
Adding item to User Menu: I don't understand the docs

### DISCUSSION THREAD

**Discussion by axelbdt** [filamentphp/filament#6354] [Help] [answered]
Hey all,

I am very new to Laravel, so the fault is probably mine. In any case let me say that I find the filament framework delightful and productive so far.

However, I sometimes lack some context to understand the docs, in this instance to [add an item to the user menu](https://filamentphp.com/docs/2.x/admin/navigation#customizing-the-user-menu).

Where do I put the code snippet from the docs:
```php
use Filament\Facades\Filament;
use Filament\Navigation\UserMenuItem;
 
Filament::serving(function () {
    Filament::registerUserMenuItems([
        UserMenuItem::make()
            ->label('Settings')
            ->url(route('filament.pages.settings'))
            ->icon('heroicon-s-cog'),
        // ...
    ]);
});
```
Should I generate a service provider with artisan and put the code in one method? Or is it the content of a full file? How is this code called?

Would appreciate some hand-holding with this :)
Cheers

**Comment by danharrin** [comment, marked as answer]
You can put it in the `boot()` method of any service provider. For example, AppServiceProvider.

**Comment by axelbdt** [comment]
Thank you, Dan :)
